### PR TITLE
Add backend filter fold option to multi_backend [JIRA: RIAK-1594]

### PR DESCRIPTION
**Pull request by nickelization**
Adds an option that can be passed to multi_backend bucket folds. The
option {backend, [BACKEND_LIST]} will cause the fold to skip over any
backends not explicitly named in the backend list.

This enhancement addresses issue #1036.

This is my first Basho PR, so apologies in advance if I've neglected any
important details! Fingers crossed :)

### CODE REVIEW

**Comment by bsparrow435** [comment]
Thank you so much for the PR! We'll get it assigned to a milestone and reviewed as soon as possible.

File: src/riak_kv_multi_backend.erl [hunk 1]
```diff
@@ -85,7 +85,7 @@
 -define(CAPABILITIES, [async_fold, index_reformat]).
 -define(ANY_CAPABILITIES, [indexes, iterator_refresh]).
 
--record (state, {backends :: [{atom(), atom(), term()}],
+-record (state, {backends :: [{atom(), atom(), term()}], % [{BackendName, BackendModule, SubState}]
```
Comment (seancribbs) [inline src/riak_kv_multi_backend.erl:88]: Wow, I didn't realize this assumed atom for the BackendName. In the [configs](https://github.com/basho/riak_kv/blob/develop/priv/multi_backend.schema#L59) they are all binaries so they can be set via bucket props.

Comment (nickelization) [inline src/riak_kv_multi_backend.erl:88]: Huh, weird. I didn't realize they could be anything other than atoms, but I guess other types should work just as well. I'm still learning the basics with Riak, but are bucket props all supposed to be binaries? From the type specs in riak_core_bucket_props, it looks like anything should be valid, and in the unit test I wrote it seems like atoms worked fine. Or, then again maybe I'm totally misinterpreting something....

**Comment by nickelization** [comment]
Just realized this never got approved. Any changes needed?

**Comment by andrewjstone** [comment]
+1 5558738

Code looks good, test passes

**Comment by andrewjstone** [comment]
@borshop merge

**Pull request merged**